src/lib/sidenav/sidenav.ts

```diff
@@ -175,53 +154,28 @@ export class MdSidenav implements AfterContentInit, OnDestroy {
   }
 
 
-  /** Open this sidenav, and return a Promise that will resolve when it's fully opened (or get
-   * rejected if it didn't). */
-  open(): Promise<MdSidenavToggleResult> {
+  /**  Open the sidenav. */
+  open(): void {
     return this.toggle(true);
   }
 
-  /**
-   * Close this sidenav, and return a Promise that will resolve when it's fully closed (or get
-   * rejected if it didn't).
-   */
-  close(): Promise<MdSidenavToggleResult> {
+  /** Close the sidenav. */
+  close(): void {
     return this.toggle(false);
   }
 
   /**
-   * Toggle this sidenav. This is equivalent to calling open() when it's already opened, or
-   * close() when it's closed.
+   * Toggle this sidenav.
    * @param isOpen Whether the sidenav should be open.
-   * @returns Resolves with the result of whether the sidenav was opened or closed.
    */
-  toggle(isOpen: boolean = !this.opened): Promise<MdSidenavToggleResult> {
-    // Shortcut it if we're already opened.
-    if (isOpen === this.opened) {
-      return this._toggleAnimationPromise ||
-          Promise.resolve(new MdSidenavToggleResult(isOpen ? 'open' : 'close', true));
-    }
+  toggle(isOpen: boolean = !this.opened): void {
```

Comment:
That won't work, because the Angular animation events don't propagate IIRC. The way to do it would be:

```
snav.onOpen.subscribe(() => doStuff());
snav.open();
```

I guess we could return the `onOpen` and `onClose` observables from the open/close methods, but I'm not sure whether it's necessary. The promises that were a part of the old API looked like they were there only there for synchronizing the animations internally.